posthog/api/test/test_dashboard.py

```diff
@@ -191,6 +193,67 @@ def test_return_cached_results(self):
         self.assertAlmostEqual(Dashboard.objects.get().last_accessed_at, now(), delta=timezone.timedelta(seconds=5))
         self.assertEqual(response["items"][0]["result"][0]["count"], 0)
 
+    def _get_dashboard_counting_queries(self, dashboard: Dashboard) -> Tuple[int, List[Dict[str, str]]]:
+        db_connection = connections[DEFAULT_DB_ALIAS]
+
+        with CaptureQueriesContext(db_connection) as capture_query_context:
+            response = self.client.get(f"/api/projects/{self.team.id}/dashboards/%s/" % dashboard.pk)
+            self.assertEqual(response.status_code, status.HTTP_200_OK)
+
+            query_count = len(capture_query_context.captured_queries)
+            return query_count, capture_query_context.captured_queries
+
+    def test_adding_insights_is_not_nplus1_for_gets(self):
+        dashboard = Dashboard.objects.create(team=self.team, name="dashboard")
+        filter_dict = {
+            "events": [{"id": "$pageview"}],
+            "properties": [{"key": "$browser", "value": "Mac OS X"}],
+            "insight": "TRENDS",
+        }
+        filter = Filter(data=filter_dict)
+
+        query_counts: List[int] = []
+        queries: List[List[Dict[str, str]]] = []
+
+        count, qs = self._get_dashboard_counting_queries(dashboard)
+        query_counts.append(count)
+        queries.append(qs)
+
+        # add insights to the dashboard and count how manh queries to read the dashboard afterwards
+        Insight.objects.create(
+            dashboard=dashboard, filters=filter_dict, team=self.team,
+        )
+        count, qs = self._get_dashboard_counting_queries(dashboard)
+        query_counts.append(count)
+        queries.append(qs)
+
+        Insight.objects.create(
+            dashboard=dashboard, filters=filter.to_dict(), team=self.team,
+        )
+        count, qs = self._get_dashboard_counting_queries(dashboard)
+        query_counts.append(count)
+        queries.append(qs)
+
+        Insight.objects.create(
+            dashboard=dashboard, filters=filter.to_dict(), team=self.team,
+        )
+        count, qs = self._get_dashboard_counting_queries(dashboard)
+        query_counts.append(count)
+        queries.append(qs)
+
+        Insight.objects.create(
+            dashboard=dashboard, filters=filter.to_dict(), team=self.team,
+        )
+        count, qs = self._get_dashboard_counting_queries(dashboard)
+        query_counts.append(count)
+        queries.append(qs)
+
+        # regression test
+        # getting a dashboard was originally n plus 1,
+        # with number of queries growing as the number of insights on the dashboard grew
+        # now a stable 8 queries no matter how many insights are on the dashboard
+        self.assertEqual(query_counts, [8, 8, 8, 8, 8], f"received: {query_counts} for queries: \n\n {queries}")
```

Comment:
Can we just assert all the counts are the same? That way we won't be tripped up if e.g. a middleware introduces another query.

Reply:
Hmmm, good point. The output in the error message is where it's useful to know what is happening